### DISCUSSION TITLE
Command to delete git-branchless data in a repository

### DISCUSSION THREAD

**Discussion by ilyagr** [arxanas/git-branchless#510] [Ideas]
I got git-branchless into a weird state (unfortunately, it seems different to reproduce), and I was looking for a way to reinitialize it from a clean slate. It turns out that `git-branchless init --uninstall` only deletes the git-branchless *configuration* added to the git repository, but not its *data*.

Firstly, I think this would be a nice feature - there should be an option to `git-branchless init --uninstall --wipe-data`, both for people who change their mind about using git-branchless and for people who want to start afresh. The documentation should mention that you should add a branch to any commit chains you want to keep before doing this.

Secondly, I came up with a command to do this that I wanted to share. I think that wipes everything, right?

```bash
git branchless init --uninstall &&
rm -r $(git rev-parse --git-path branchless) $(git rev-parse --git-path refs/branchless) 
```

**Comment by arxanas** [comment]
> Secondly, I came up with a command to do this that I wanted to share. I think that wipes everything, right?

Yep, that looks about right. I didn't make a command which deletes data since I hadn't needed it and I worried about deleting important data, but it probably makes sense to create one at this point. Feel free to open a pull request 😄.

> I got git-branchless into a weird state (unfortunately, it seems different to reproduce)

You can also try `git-branchless repair` to fix certain issues.

**Reply by ilyagr** [reply]
👍 , thanks!
